pymongo/lock.py

```diff
@@ -39,209 +52,40 @@ def _create_lock() -> threading.Lock:
     return lock
 
 
+def _async_create_lock() -> Lock:
+    """Represents an asyncio.Lock."""
+    return Lock()
+
+
+def _create_condition(
+    lock: threading.Lock, condition_class: Optional[Any] = None
+) -> threading.Condition:
+    """Represents a threading.Condition."""
+    if condition_class:
```

Comment:
Sigh, I can't believe we still have to write code for condition_class even though it's never used. I reopened https://jira.mongodb.org/browse/PYTHON-1149.

Reply:
Yeah I wasn't sure if we actually still used it, removing it entirely would be great.